lambda/session/lambda_functions.py

```diff
@@ -142,11 +194,65 @@ def delete_user_sessions(event: dict, context: dict) -> Dict[str, bool]:
     logger.info(f"Deleting all sessions for user {user_id}")
     sessions = _get_all_user_sessions(user_id)
     logger.debug(f"Found user sessions: {sessions}")
-    for session in sessions:
-        _delete_user_session(session["sessionId"], user_id)
+
+    with ThreadPoolExecutor() as executor:
+
+        def delete_session_process(session: dict) -> None:
+            try:
+                session_id = session["sessionId"]
+                logger.info(f"Deleting session with ID {session_id} for user {user_id}")
+                _delete_user_session(session_id, user_id)
+            except Exception as e:
+                print(f"Error deleting session: {e}")
+
+        list(executor.map(delete_session_process, sessions))
     return {"deleted": True}
 
 
+@api_wrapper
+def attach_image_to_session(event: dict, context: dict) -> dict:
+    """Append the message to the record in DynamoDB."""
+    try:
+        session_id = get_session_id(event)
+
+        try:
+            body = json.loads(event["body"], parse_float=Decimal)
+        except json.JSONDecodeError as e:
+            return {"statusCode": 400, "body": json.dumps({"error": f"Invalid JSON: {str(e)}"})}
+
+        if "message" not in body:
+            return {"statusCode": 400, "body": json.dumps({"error": "Missing required fields: messages"})}
+
+        message = body["message"]
+
+        if (
+            message.get("type", None) == "image_url"
+            and message.get("image_url", {}).get("url", None) is not None
```

Comment:
`message.get("image_url", {})` is repeated quite a bit in this function. I think pulling out some of the more common things up after `message = body["message"]` would clean up the code and make this if statement easier to understand.